### ISSUE TITLE
Automatic fetch at start up and periodically

### ISSUE THREAD

**Issue by MGS-SEBASTIEN** [github/VisualStudio#2207]
Hi,

I don't know if something like that has already been asked but I didn't find anything on it.

As all knows, Git Hub is useful as collaborative work, but... by trying GitHub extension in visual studio I found some new features that could be really nice to increase collaborative working experience.

Is it possible to put three options in GitHub extension options ?
1) Automatic fetch when we open a project or website or folder in VS (a checkbox option)
2) Automatic fetch when we switch branch (a checkbox option).
3) Autmatic fetch on the actual branch every X minutes (a textbox option)

I think you understood what I mean, colleagues could now know when something has been modified on the branch they are working.

**Comment by jcansdale** [comment]
@MGS-SEBASTIEN,

I agree something like this would be nice. Not having the latest commits locally is something that regularly trips me up.

There is an extension by @madskristensen that tackles your first suggestion (it's currently VS 2019 only).

I started working on a PR to add VS 2017 support and make it update all branches (not just the current one).
https://github.com/madskristensen/GitPull/pull/3

If you're interested in trying this version, you can find the .vsix here:
https://ci.appveyor.com/project/madskristensen/gitpull/builds/21339372/artifacts

I'm a little hesitant to do thing anything automatically that make changes in the users working directory. I wonder is simply having a notification when there are remote chances would make more sense? The user could then click on the notification, see the incoming changes and update their repository. That way they'd always be aware of the latest commits.

**Comment by Defee** [comment]
Hi! 
@jcansdale the problem is that git pull extension by @meaghanlewis performs pull which is not the case the person is asking. 

As you know pull do the following:
1. fetches 
1. merges 
So I would've preferred to have extensions doing the fetch instead of the pull. 

Fetch is usually used in the git workflows and it makes really big problem for some people to go and fetch repo, which leads in their confusion not seeing the changes from remote.

Notification make sense, but I would've preferred a checkbox enabling me to do auto-fetches. I'm using the same feature in vscode ([link](https://code.visualstudio.com/docs/editor/versioncontrol#_remotes)) and it is awesome cos it reduces a lot of hassle.

**Comment by devel112** [comment]
Just to be clear, pulling is entirely out of the questing. This is only about auto-fetching.

**Comment by mselley** [comment]
would be good to see this, can trip over dev especially when they start out or are used to TFVC.

**Comment by joeamnesiac** [comment]
I would like to see this brought into Visual Studio. I already use it in VSCode and find it very useful.

**Comment by charlesroddie** [comment]
Until then try [GitAutoFetch](https://marketplace.visualstudio.com/items?itemName=ZweiDeveloper.gitAutoFetch)

**Comment by jfranki** [comment]
As far as I know this is the default in VS Code, why is not even configurable in Visual Studio?
